ServerCore/Pages/Teams/CreateTeam.razor.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Forms;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using ServerCore.DataModel;
+using ServerCore.Helpers;
+using ServerCore.ModelBases;
+using System.Linq;
+
+namespace ServerCore.Pages.Teams
+{
+    class EmailListValidationAttribute : ValidationAttribute
+    {
+        protected override ValidationResult IsValid(object value, ValidationContext validationContext)
+        {
+            string emailList = value as string;
+            if (string.IsNullOrWhiteSpace(emailList))
+            {
+                return new ValidationResult("An email is required.");
+            }
+
+            if (!MailHelper.IsValidEmail(emailList))
+            {
+                return new ValidationResult("Invalid primary contact email.");
+            }
+
+            return ValidationResult.Success;
+        }
+    }
+
+    class TeamUniquenessValidationAttribute : ValidationAttribute
+    {
+        protected override ValidationResult IsValid(object value, ValidationContext validationContext)
+        {
+            PuzzleServerContext puzzleServerContext = validationContext.GetRequiredService<PuzzleServerContext>();
+            TeamModel teamModel = (TeamModel)validationContext.ObjectInstance;
+            string name = value as string;
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                return new ValidationResult("Team names cannot be left blank.");
+            }
+
+            if (TeamHelper.IsTeamNameTaken(puzzleServerContext, teamModel.EventId, name))
+            {
+                return new ValidationResult("Another team has this name.");
+            }
+
+            return ValidationResult.Success;
+        }
+    }
+
+    public class TeamModel
```

Comment:
Assuming this is supposed to mirror the same properties in the server-side Team data model, it would be nice to add a comment here and in that file saying so.  That way people know that if they ever modify one of the fields here, they need to do it there too, and vice versa.

Reply:
Done